src/artm/core/instance.cc

```diff
@@ -381,6 +398,18 @@ void Instance::DisposeDictionary(const std::string& name) {
   dictionaries_.erase(name);
 }
 
+void Instance::DisposeDictionaryImpl(const std::string& name) {
```

Comment:
I think the code would be cleaner if you remove this method and move `dictionaries_impl_.erase(name)` into `Instance::DisposeDictionary` method. This would be similar to how merger handles old- and new-style models in `Merger::DisposeModel` method.
``` 
  topic_model_.erase(model_name);
  phi_matrix_.erase(model_name);
```